### PR TITLE
feat: add utxos double check for all btc related flows

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of Leather — [Extension build](https://github.com/leather-wallet/extension/actions/runs/7959596534), [Test report](https://leather-wallet.github.io/playwright-reports/feat/prevent-all-flows)

Refactored utxos double check, so it would be much easier to use it in broadcast tx flow

### CODE REVIEW

File: src/app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts [hunk 1]
```diff
@@ -19,20 +25,29 @@ export function useBitcoinBroadcastTransaction() {
...
       delayTime = 700,
-      checkForInscribedUtxos,
     }: BroadcastCallbackArgs) => {
       try {
-        // To-Do implement check in all btc related flows and remove conditional usage
-        // add explicit check in broadcastTx to ensure that utxos are checked before broadcasting
-        const hasInscribedUtxos = await checkForInscribedUtxos?.();
+        let inputs: btc.TransactionInput[] = decodeBitcoinTx(tx).inputs;
+
+        // Filer out intentional spend inscription txid from the check list
```
Comment (fbwoolf) [inline src/app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts]: Just a nit here in the comment: `Filter...`

**Review by fbwoolf — APPROVED** [review]

File: src/app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts [hunk 2]
```diff
@@ -19,20 +25,29 @@ export function useBitcoinBroadcastTransaction() {
...
-        const hasInscribedUtxos = await checkForInscribedUtxos?.();
+        let inputs: btc.TransactionInput[] = decodeBitcoinTx(tx).inputs;
+
+        // Filter out intentional spend inscription txid from the check list
+        if (sendingInscriptionsList.length > 0) {
+          inputs = filterOutIntentionalInscriptionsSpend({
+            inputs: decodeBitcoinTx(tx).inputs,
+            inscriptions: sendingInscriptionsList,
+          });
+        }
```
Comment (kyranjamie) [inline src/app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts]: Anyway we could write this without reassigning the variable? Why not just use `filter`?

**Review by kyranjamie — APPROVED** [review]

File: src/app/pages/send/ordinal-inscription/send-inscription-review.tsx [hunk 3]
```diff
@@ -45,17 +40,9 @@ export function SendInscriptionReview() {
...
-    inputs: filterOutIntentionalInscriptionsSpend({
-      inputs: decodeBitcoinTx(bytesToHex(signedTx)).inputs,
-      inscriptions: [inscription],
-    }),
-  });
-
   async function sendInscription() {
     await broadcastTx({
-      checkForInscribedUtxos: checkIfUtxosListIncludesInscribed,
+      sendingInscriptionsList: [inscription],
```
Comment (kyranjamie) [inline src/app/pages/send/ordinal-inscription/send-inscription-review.tsx]: How about a more generic `skipBypassUtxoCheckFor: utxo[]`?

File: src/app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts (hunk 2 above)
Comment (alter-eggo) [reply src/app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts]: yeah, agree here

File: src/app/pages/send/ordinal-inscription/send-inscription-review.tsx (hunk 3 above)
Comment (alter-eggo) [reply src/app/pages/send/ordinal-inscription/send-inscription-review.tsx]: changed to pass just ids `skipBypassUtxoIdsCheckFor`

**Pull request merged**